pkg/cache/cache.go

```diff
@@ -86,3 +90,148 @@ func (c *Cache[T]) IsDAIncluded(hash string) bool {
 func (c *Cache[T]) SetDAIncluded(hash string) {
 	c.daIncluded.Store(hash, true)
 }
+
+const (
+	itemsByHeightFilename = "items_by_height.gob"
+	itemsByHashFilename   = "items_by_hash.gob"
+	hashesFilename        = "hashes.gob"
+	daIncludedFilename    = "da_included.gob"
+)
+
+// saveMapGob saves a map to a file using gob encoding.
+func saveMapGob[K comparable, V any](filePath string, data map[K]V) error {
```

Comment:
what is the difference with using this and the underlying database?

Reply:
I found it easier for cleaning the cache. As it is read only twice, but we go over many keys.